### PR TITLE
feat(resource/cache): current weapon

### PR DESCRIPTION
**Pull request by BerkieBb**
may as well

### CODE REVIEW

File: resource/cache/client.lua [hunk 1]
```diff
@@ -37,6 +37,16 @@ CreateThread(function()
...
 		end
 
+		local hasWeapon, currentWeapon = GetCurrentPedWeapon(ped, true)
+
+		if hasWeapon then
+			if not cache.weapon or cache.weapon ~= currentWeapon then
+				cache:set('weapon', currentWeapon)
+			end
+		else
+			cache:set('weapon', false)
```
Comment (thelindat) [inline resource/cache/client.lua:47]: Don't you technically always have a weapon (i.e. weapon_unarmed)?

File: resource/cache/client.lua [hunk 2]
```diff
@@ -37,6 +37,16 @@ CreateThread(function()
 			cache:set('seat', false)
 		end
 
+		local hasWeapon, currentWeapon = GetCurrentPedWeapon(ped, true)
+
+		if hasWeapon then
+			if not cache.weapon or cache.weapon ~= currentWeapon then
```
Comment (thelindat) [inline resource/cache/client.lua:43]: `cache.weapon ~= currentWeapon` should be enough 🤷

**Review by thelindat — CHANGES_REQUESTED** [review]

File: resource/cache/client.lua (hunk 1 above)
Comment (BerkieBb) [reply resource/cache/client.lua:47]: If your currentWeapon is weapon_unarmed, hasWeapon will be false

Comment (BerkieBb) [reply resource/cache/client.lua:47]: So actually reading that again, the if statement is probably not even needed at all

**Comment by BerkieBb** [comment]
Accidentally stopped thinking

**Comment by BerkieBb** [comment]
Nevermind, that commit was actually good as I just now saw that cache:set already checks for the previous value, so here comes the double revert

**Review by LukeWasTakenn — APPROVED** [review]

**Pull request merged**